src/main/java/zaasclient/HttpsClient.java

```diff
@@ -0,0 +1,201 @@
+     * @throws UnrecoverableKeyException
+     * @throws KeyManagementException
+     */
+    public CloseableHttpClient getHttpsClientWithKeyStoreAndTrustStore()
+        throws NoSuchAlgorithmException, IOException, KeyStoreException, CertificateException, UnrecoverableKeyException, KeyManagementException {
+        SSLContext sslContext;
+        TrustManagerFactory tmf = null;
+        KeyManagerFactory kmf = null;
+
+        if (trustStorePath != null) {
+            tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            KeyStore trustStore = KeyStore.getInstance(trustStoreType);
+            File trustFile = new File(trustStorePath);
+            trustStore.load(new FileInputStream(trustFile), trustStorePassword.toCharArray());
+            tmf.init(trustStore);
+        }
+
+        if (keyStorePath != null) {
+            kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            KeyStore keyStore = KeyStore.getInstance(keyStoreType);
+            File keyFile = new File(keyStorePath);
+            keyStore.load(new FileInputStream(keyFile), keyStorePassword.toCharArray());
+            kmf.init(keyStore, keyStorePassword.toCharArray());
+        }
+
+        sslContext = this.getSSLContext(kmf, tmf);
+
+        return getCloseableHttpClient(sslContext);
+    }
+
+    private CloseableHttpClient getCloseableHttpClient(SSLContext sslContext) {
+        final SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(sslContext,
+            SSLConnectionSocketFactory.getDefaultHostnameVerifier());
+
+        closeableHttpsClient = HttpClients.custom()
+            .setSSLSocketFactory(sslsf)
+            .setDefaultRequestConfig(this.requestConfig)
+            .setMaxConnTotal(3 * 3)
+            .setMaxConnPerRoute(3)
+            .build();
+        return closeableHttpsClient;
+    }
+
+    /**
+     * Overloaded method of the getclosableHttpClient method .. so that the cookies can be used for storing jwt tokens
+     *
+     * @param sslContext
+     * @param cookieStore
+     * @return
+     */
+    private CloseableHttpClient getCloseableHttpClient(SSLContext sslContext, BasicCookieStore cookieStore) {
+        final SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(sslContext,
+            SSLConnectionSocketFactory.getDefaultHostnameVerifier());
+
+        closeableHttpsClient = HttpClients.custom()
+            .setSSLSocketFactory(sslsf)
+            .setDefaultRequestConfig(this.requestConfig)
+            .setMaxConnTotal(3 * 3)
+            .setMaxConnPerRoute(3)
+            .setDefaultCookieStore(cookieStore)
+            .build();
+        return closeableHttpsClient;
+    }
+
+
+    /**
+     * @param kmf
+     * @param tmf
+     * @return
+     * @throws Exception
+     */
+    private SSLContext getSSLContext(KeyManagerFactory kmf, TrustManagerFactory tmf) throws KeyManagementException, NoSuchAlgorithmException {
```

Comment:
As it's described above, after the refactoring there will be two methods instead this one
```
private getSSLContextWithTrustStore() {
  SSLContext sslContext = SSLContext.getInstance("TLS");
  sslContext.init(null, getTrustManagersOrNull(), new SecureRandom());
  return sslContext;
}

private getSSLContextWithKeyStoreAndTrustStore() {
  SSLContext sslContext = SSLContext.getInstance("TLS");
  sslContext.init(getKeyManagersOrNull(), getTrustManagersOrNull(), new SecureRandom());
  return sslContext;
}
```